### DISCUSSION TITLE
Infer the type of Enumerable.Empty and similar constructs based on return types

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#3926] [General] [answered]
If I have a method like so:

```
public IEnumerable<int> GetEmptyNumberList()
{
    return Enumerable.Empty();
}
```

I get a compile error saying that the type parameter of `Enumerable.Empty()` cannot be inferred. It would be nice if this were possible so we don't have to go duplicating the return type.

**Comment by PathogenDavid** [comment, marked as answer]
This has been requested before: https://github.com/dotnet/csharplang/discussions/92
